Tasks/PublishSymbolsV2/PreJobExecutionPublishSymbols.ts

```diff
@@ -0,0 +1,42 @@
+const path = require('path');
+const clientToolUtils = require('clienttool-common/ClientToolUtilities');
+const tl = require('azure-pipelines-task-lib');
+
+const toolName = "symbol";
+
+async function PreJobExecutionPublishSymbols(){
+    tl.setResourcePath(path.join(__dirname, "task.json"));
+    const UseNetCoreClientTool = tl.getBoolInput("UseNetCoreClientTool", false);
+
+    try {
+        // feature flag true or none Windows, then download NetCore version of client tool
+        if (UseNetCoreClientTool || tl.osType() != "Windows_NT") {
+
+            // Getting NetCore client tool
+            tl.debug("Getting NetCore client tool");
+            let clientToolFilePath = "";
+
+            // Downloading the correct version of Symbol to target location
+            const accessToken = clientToolUtils.getSystemAccessToken();
+            const serviceUri = tl.getEndpointUrl("SYSTEMVSSCONNECTION", false);
+            const blobUri = await clientToolUtils.getBlobstoreUriFromBaseServiceUri(serviceUri, accessToken);
+
+            tl.debug(tl.loc("Info_RetrievingClientToolUri", blobUri));
+
+            // Downloading the client tool
+            clientToolFilePath = await clientToolUtils.retryOnExceptionHelper(
+                () => clientToolUtils.getClientToolFromService(
+                    blobUri,
+                    accessToken,
+                    toolName), 3, 1000);
+
+            tl.setTaskVariable('CLIENTTOOL_FILE_PATH', clientToolFilePath);
+        }
+
+    } catch (error) {
+        tl.error(error);
+        tl.setResult(tl.TaskResult.Failed, tl.loc("PreJobFailedToExecution"));
```

Comment:
This should probably be "PreJobFailedToExecute" and a corresponding string should be added to the "messages" section of task.json